tenant/repository.go

```diff
@@ -162,35 +193,69 @@ func (s DBService) DeleteTenant(tenantID uuid.UUID) error {
 	return s.db.Unscoped().Delete(&Tenant{ID: tenantID}).Error
 }
 
-type NilService struct {
+func (s DBService) NewTenantRepository(tenantID uuid.UUID) Repository {
+	return DBTenantRepository{service: s, tenantID: tenantID}
+}
+
+type Repository interface {
+	GetTenant() (*Tenant, error)
+	NewNamespace(envType environment.Type, nsName, masterURL string, state NamespaceState) *Namespace
+	GetNamespaces() ([]*Namespace, error)
+	SaveNamespace(namespace *Namespace) error
+	CreateNamespace(namespace *Namespace) (*Namespace, error)
+	DeleteNamespace(namespace *Namespace) error
+	DeleteNamespaces() error
+	DeleteTenant() error
+	Service() Service
+}
+
+type DBTenantRepository struct {
+	service  DBService
+	tenantID uuid.UUID
+}
+
+func (n DBTenantRepository) NewNamespace(envType environment.Type, nsName, masterURL string, state NamespaceState) *Namespace {
+	return &Namespace{
+		TenantID:  n.tenantID,
+		Name:      nsName,
+		Type:      envType,
+		State:     state,
+		MasterURL: masterURL,
+	}
+}
+
+func (n DBTenantRepository) GetTenant() (*Tenant, error) {
+	return n.service.GetTenant(n.tenantID)
 }
 
-func (s NilService) Exists(tenantID uuid.UUID) bool {
-	return false
+func (n DBTenantRepository) GetNamespaces() ([]*Namespace, error) {
+	return n.service.GetNamespaces(n.tenantID)
 }
 
-func (s NilService) GetTenant(tenantID uuid.UUID) (*Tenant, error) {
-	return nil, nil
+func (n DBTenantRepository) SaveNamespace(namespace *Namespace) error {
+	namespace.TenantID = n.tenantID
+	return n.service.SaveNamespace(namespace)
 }
 
-func (s NilService) GetNamespaces(tenantID uuid.UUID) ([]*Namespace, error) {
-	return nil, nil
+func (n DBTenantRepository) CreateNamespace(namespace *Namespace) (*Namespace, error) {
+	namespace.TenantID = n.tenantID
+	return n.service.CreateNamespace(namespace)
```

Comment:
same here: no need to have this method here?

Reply:
should be solved as part of 1c4a496

Reply:
Thanks!